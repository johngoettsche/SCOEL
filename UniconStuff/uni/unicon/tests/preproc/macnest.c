/* test nesting of directives within multi-line macros */
#begdef foo(y,z)
   #undef x
   #begdef x
      #undef z##1
      #begdef z##1
         01
         02
         03
      #enddef
      y
   #enddef
#enddef
x
a1
b1
foo(1,a)
x
a1
b1
foo(2,b)
x
a1
b1
#begdef include(s)
#include #s
#enddef
include(macnest.h)

#begdef conditionals(a)
#if a
a_true
#else
a_false
#endif
#enddef
#define n 1
conditionals(n)
#undef n
#define n 0
conditionals(n)
