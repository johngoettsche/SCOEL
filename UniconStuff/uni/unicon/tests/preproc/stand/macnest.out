#line 13 "macnest.c"

x
a1
b1

#line 10 "macnest.c"

      1
   
#line 6 "macnest.c"

         01
         02
         03
      
#line 19 "macnest.c"

b1

#line 10 "macnest.c"

      2
   
#line 6 "macnest.c"

         01
         02
         03
      
#line 6 "macnest.c"

         01
         02
         03
      
#line 1 "macnest.h"
"macnest.h"


#line 31 "macnest.c"

a_true

a_false

